src/gdsync.cpp

```diff
@@ -1389,6 +1387,15 @@ struct gds_qp *gds_create_qp(struct ibv_pd *pd, struct ibv_context *context, gds
                 return NULL;
         }
 
+        gds_dbg("before gds_register_peer_ex\n");
+        ret = gds_register_peer_ex(context, gpu_id, &peer, &peer_attr);
+        if (ret) {
+                gds_err("error %d in gds_register_peer_ex\n", ret);
+                goto err_free_cqs;
+        }
+        peer->res_domain = gds_create_res_domain(context);
+        gds_dbg("created res_domain=%p for GPU %u\n", peer->res_domain, gpu_id);
+
         gds_dbg("creating TX CQ\n");
         tx_cq = gds_create_cq(context, qp_attr->cap.max_send_wr, NULL, NULL, 0, gpu_id, 
```

Comment:
let's introduce an internal version of gds_create_cq which takes a peer as argument, and use it here.
gds_create_cq() will be changed to call gds_register_peer_ex() and then that new internal version